### PR TITLE
clean: fix compiler warnings about QTranslator

### PR DESCRIPTION
**Pull request by shenlebantongying**
Handling errors religiously

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -564,13 +564,21 @@ int main( int argc, char ** argv )
   QLocale::setDefault( locale );
   QApplication::setLayoutDirection( locale.textDirection() );
 
-  if ( !qtTranslator.load( "qt_extra_" + localeName, Config::getLocDir() ) ) {
-    qtTranslator.load( "qt_extra_" + localeName, QLibraryInfo::location( QLibraryInfo::TranslationsPath ) );
+  if ( qtTranslator.load( "qt_extra_" + localeName, Config::getLocDir() )
+       || qtTranslator.load( "qt_extra_" + localeName, QLibraryInfo::path( QLibraryInfo::TranslationsPath ) ) ) {
```
Comment (shenlebantongying) [inline src/main.cc:568]: The 2nd condition won't be evaluated if the first one success.

https://en.wikipedia.org/wiki/Short-circuit_evaluation

**Comment by shenlebantongying** [comment]
BTW, do we still needs `qt_extra` which was added 11 years ago to fix Qt's problem!

https://github.com/xiaoyifang/goldendict-ng/blame/staged/locale/qt_extra_it.ts

**Comment by xiaoyifang** [comment]
need to confirm and compare with qbase's own `it`(italian?) language translation ,qt_it.qm or qtbase_it.qm?

**Comment by shenlebantongying** [comment]
The one we have is not updated for 10 years and likely won't be updated forever. Let's just use the Qt's one and hope it improves as it should.

**Pull request closed without merging**